src/app/patient/patient.cpp

```diff
@@ -0,0 +1,83 @@
+#include "patient.hpp"
+
+std::vector<std::shared_ptr<Patient>> Patient::patients_;
+
+Patient::Patient(const std::string name, const std::string last_name, const std::string pesel,
+                 const std::unordered_set<Allergent> allergents)
+    : Person(name, last_name, pesel), allergents_{allergents}, debt_{}
+{
+}
+
+void Patient::createPatient(const std::string name, const std::string last_name, const std::string pesel,
+                            const std::unordered_set<Allergent> allergents)
+{
+    auto patient = std::shared_ptr<Patient>(new Patient(name, last_name, pesel, allergents));
+    patients_.push_back(patient);
+}
```

Comment:
Using static function to initialise objects should be only done when we use composition. In our case in standard association, consider making destructor private.

Reply:
We have to decide when an object (here Patient) should be added to the extent.
If we will create patient at the time when visit occures then it's okay to remove it.
However we need to have lists of patients even tho they do not have visit or visit is not pointing to any of patients. 
In this case we need to append extent and there's an issue, that we cannot do that at construction time if we want to use smart pointers. 
Nor smart pointer, nor enable_shared_from_this allows us to do that, because there's memory allocated yet, we need to wait until construction is finished.
Solving this issue is simple with create method, where we create an object, assign it's pointer to extent. 
I would consider changing documentation to mark, that we need to perform such operation due to language choice of our.

Reply:
If you find more elegant way to do that I would really appreciate

Reply:
We spoke that we would need to hide dtor, but that's not possible while we use smart pointers. 
Dtor should stay default as it is and if we want to perform deletion by hand (ex. removing patient), it is possible via removePatient method call. It's use it shown within tests.